src/foam/nanos/auth/RetrievePassword.js

```diff
@@ -66,7 +67,7 @@ foam.CLASS({
         return ! errors_;
       },
       code: function(X) {
-        const user = this.User.create({ email: this.email });
+        const user = this.User.create({ email: this.identifier });
```

Comment:
if this is an identifier - how do we know to set it on the email? Its probably best that add another property for userName and set both on this created user. In the emailToken service you can validate and check if the criteria is met...as long as the errors here are caught and sent to the user by ..